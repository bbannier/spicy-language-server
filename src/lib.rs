#![feature(str_strip)]

#[macro_use]
extern crate pest;

pub mod lsp;
pub mod options;
pub mod spicy;
