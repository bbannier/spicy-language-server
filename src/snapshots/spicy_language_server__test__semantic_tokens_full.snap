---
source: src/lib.rs
expression: server.initialize.capabilities.semantic_tokens_provider
---
Some(
    SemanticTokensOptions(
        SemanticTokensOptions {
            work_done_progress_options: WorkDoneProgressOptions {
                work_done_progress: None,
            },
            legend: SemanticTokensLegend {
                token_types: [
                    SemanticTokenType(
                        "comment",
                    ),
                    SemanticTokenType(
                        "function",
                    ),
                    SemanticTokenType(
                        "variable",
                    ),
                    SemanticTokenType(
                        "keyword",
                    ),
                    SemanticTokenType(
                        "type",
                    ),
                    SemanticTokenType(
                        "property",
                    ),
                    SemanticTokenType(
                        "macro",
                    ),
                    SemanticTokenType(
                        "event",
                    ),
                    SemanticTokenType(
                        "punctuation",
                    ),
                    SemanticTokenType(
                        "number",
                    ),
                    SemanticTokenType(
                        "regexp",
                    ),
                    SemanticTokenType(
                        "string",
                    ),
                    SemanticTokenType(
                        "modifier",
                    ),
                ],
                token_modifiers: [],
            },
            range: None,
            full: Some(
                Bool(
                    true,
                ),
            ),
        },
    ),
)
